eclipse-plugin/sapl-eclipse-ui/src/main/java/io/sapl/grammar/ui/labeling/SAPLDescriptionLabelProvider.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (C) 2017-2024 Dominic Heutelbeck (dominic@heutelbeck.com)
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/*
+ * SAPL generated by Xtext
+ */
+package io.sapl.grammar.ui.labeling;
+
+import org.eclipse.xtext.ui.label.DefaultDescriptionLabelProvider;
+
+/**
+ * Provides labels for IEObjectDescriptions and IResourceDescriptions.
+ *
+ * See
+ * https://www.eclipse.org/Xtext/documentation/310_eclipse_support.html#label-provider
+ */
+public class SAPLDescriptionLabelProvider extends DefaultDescriptionLabelProvider {
+
+    // Labels and icons can be computed like this:
```

Comment:
Commented out code. Here I understand that this is a TODO to implement providing icons to Eclipse. Is this on the menu?

Reply:
From my understanding this class is for providing custom icons and labels for the outline and some other places.
I do not plan to add custom icons to the outline at the moment. The default implementation here might be enough, even if we decide to add some custom icons later.

I will try to change the icon for sapl policy files to the sapl icon, but this probable needs to be changed in the plugin.xml instead of here. (see https://www.eclipse.org/forums/index.php/t/1109080/)

This is probably the right place to customize labels in the outline and may be needed to fix one of the known issues. But I will try to fix the missing names in the outline in sapl-lang and sapl-ide first.